libdirections-onboard/src/main/java/com/mapbox/navigation/route/onboard/MapboxOnboardRouter.kt

```diff
@@ -97,16 +108,38 @@ class MapboxOnboardRouter : Router {
             )
         ).build()
 
-        OfflineRouteRetrievalTask(navigatorNative, logger, object : OnOfflineRouteFoundCallback {
-            override fun onRouteFound(routes: List<DirectionsRoute>) {
-                callback.onResponse(routes)
+        retrieveRoute(offlineRouter.buildUrl(), callback)
+    }
+
+    override fun cancel() {
+        scope.cancel()
+    }
+
+    private fun retrieveRoute(url: String, callback: Router.Callback) {
+        scope.launch {
+            val routerResult = withContext(Dispatchers.Default) {
+                synchronized(navigatorNative) {
```

Comment:
I believe we don't need `synchronized` anymore. Could you confirm @vmihaylenko @mskurydin?

Reply:
@mskurydin confirmed that it's safe to remove `synchronized` @korshaknn

Reply:
done